src/main/fc/fc_msp.c

```diff
@@ -2055,10 +2059,17 @@ static mspResult_e mspCommonProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 #endif
             } else {
 #if defined(OSD)
-                // set a position setting
-                const uint16_t pos  = sbufReadU16(src);
-                if (addr < OSD_ITEM_COUNT) {
-                    osdConfigMutable()->item_pos[addr] = pos;
+                const uint16_t value = sbufReadU16(src);
+
+                /* Get screen index, 0 is post flight statsitsics, 1 and above are in flight OSD screens */
+                const uint8_t screen = (sbufBytesRemaining(src) == 1) ? sbufReadU8(src) : 1;
```

Comment:
to be future-proof, read if enough bytes are available (`(sbufBytesRemaining(src) >= 1)`)